scripts/PROTOCOL.md

```diff
@@ -4,80 +4,67 @@ Genume traverses the files under `script/` in alphabetical order every time it n
+Commands are sent to the host by writing to stdout. Responses, if specified by the command are sent to the stdin of the child process. Stderr gets redirected to host's stdout.
 
-3. `GROUP [BAS|ADV] name`
-    - Defines a new group. All following commands will refer to the new group untill a new group or path command.
+```sh
+CONF id [dependencies...]
+SET DESCRIPTION value
+VALUE [BAS|ADV] [SUBCAT path] key value...
+SUBCAT [BAS|ADV] path
+```
 
-4. `PATH [BAS|ADV] path`
-    - Equivalent to the group command except the given value correspnds to a path of groups. The groups in the path are separated with `.`
+1. `CONF [dependencies...]`
+    - Configures the child process. This must be sent only once and be the first command sent.
+    - `dependencies` is a variable size set of strings of all the external executables the child needs. One custom dependency is `root`, which provides root access to the child(if allowed by the user).
+    - This commands replies with `OK` if everything has been setup correctly. Else it replies with an error message.
 
-5. `[VALUE|VALUES] [BAS|ADV] key value [GROUP|PATH] path`
-    - A combination of the value and group commands
-    - The group or path specified for this command will not affect the folloing commands.
+2. `SET DESCRIPTION value`
+    - Changes a property to a new value.
+    - Currently supported properties are:
+        1. `DESCRIPTION`: The description of the current (sub)category (string).
 
-#### Notes
+3. `VALUE [BAS|ADV] [SUBCAT path] key value...`
+    - Creates a new entry containing a string or multiples.
+    - `VALUE` is the command name.
+    - `[BAS|ADV]` is an optional enum. It is either `BAS` for **basic** or information or `ADV` for **advanced** information. It has no effect if the entry already exists.
+    - `[SUBCAT path]` is an optional subcategory. Unlike the full command the effects of this option are temporary.
+    - `key` is the key. It should contain only characters you would use to name a variable _(aka alphanumeric and underscores)_.
+    - `value...` is the string or strings to display. Each one must be between double or single quotation marks or parenthesis.
 
-- The values will be trimmed of whitespace.
+4. `SUBCAT [BAS|ADV] path`
+    - Creates a new subcategory or switches to an already existing one. All following commands will refer to the new subcategory until a new subcat command.
+    - `SUBCAT` the command name.
+    - `[BAS|ADV]` is an optional enum. It is either `BAS` for **basic** or information or `ADV` for **advanced** information. It has no effect if the category already exists.
+    - `path` is the path of the subcategory to switch to. Two subcategories must be seperated by a dot and should only contain alphanumeric and underscores. Paths starting with dot (.) are interpreted as relative paths. An empty path is interpreted as the master/root category.
 
-### Example script
+#### Notes
 
-```sh
```

Comment:
Leave the example, examples are always good

Reply:
Examples go with bash helpers which will get implemented at a later patch.